account_payment_purchase/models/purchase_order.py

```diff
@@ -46,18 +46,36 @@ def _get_default_supplier_partner_bank(self, partner):
     def _compute_payment_mode(self):
         for order in self:
             if order.partner_id:
+                order = order.with_company(order.company_id)
+                order.payment_mode_id = order.partner_id.supplier_payment_mode_id
+            else:
+                order.payment_mode_id = False
+
+    @api.depends("partner_id", "company_id", "payment_mode_id")
+    def _compute_supplier_partner_bank(self):
+        for order in self:
+            if (
+                order.payment_mode_id.payment_method_id.bank_account_required
+                and order.partner_id
+            ):
                 order = order.with_company(order.company_id)
                 order.supplier_partner_bank_id = (
                     order._get_default_supplier_partner_bank(order.partner_id)
                 )
-                order.payment_mode_id = order.partner_id.supplier_payment_mode_id
             else:
                 order.supplier_partner_bank_id = False
-                order.payment_mode_id = False
 
     def _prepare_invoice(self):
-        """Leave the bank account empty so that account_payment_partner set the
-        correct value with compute."""
-        invoice_vals = super()._prepare_invoice()
-        invoice_vals.pop("partner_bank_id")
-        return invoice_vals
+        res = super()._prepare_invoice()
+        if self.env.context.get("skip_payment_mode_propagation"):
+            return res
+        payment_mode_id = (
+            self.payment_mode_id or self.partner_id.supplier_payment_mode_id
```

Comment:
Is it correct that in case there's no a payment mode set in the purchase order the invoice is created with the payment mode set in the vendor? Shouldn't it be left empty in the invoice too so the information matches?

Reply:
You are right, changed!